core/src/view/view.cpp

```diff
@@ -68,12 +68,48 @@ void View::setSize(int _width, int _height) {
     // Screen space orthographic projection matrix, top left origin, y pointing down
     m_orthoViewport = glm::ortho(0.f, (float)m_vpWidth, (float)m_vpHeight, 0.f, -1.f, 1.f);
 
+    // If the viewport dimention change and the view trapezoid is not within the mapbound,
+    // zoom in to get view trapezoid within map bounds
+    if(!checkMapBound()) { m_zoom_prev += 1; }
+
+}
+
+bool View::checkMapBound() {
+
+    auto mapBounds = m_projection->MapBounds();
+
+    // Make sure world space view trapezoid bounds are within mapBounds
+    // If any of the 4 view trapezoid bounds are off the mapBounds, do not set the view Position
+    glm::dvec2 bottomLeft = { 0.f, m_vpHeight};
+    glm::dvec2 bottomRight = { m_vpWidth, m_vpHeight};
+    glm::dvec2 topRight = { m_vpWidth, 0.f};
+    glm::dvec2 topLeft = { 0.f, 0.f};
+
+    screenToGroundPlane(bottomLeft.x, bottomLeft.y);
+    bottomLeft += glm::dvec2(m_pos.x, m_pos.y);
+    if ( bottomLeft.x < mapBounds.min.x || bottomLeft.x > mapBounds.max.x ||
+         bottomLeft.y < mapBounds.min.y || bottomLeft.y > mapBounds.max.y ) { return false; }
+    screenToGroundPlane(bottomRight.x, bottomRight.y);
+    bottomRight += glm::dvec2(m_pos.x, m_pos.y);
+    if ( bottomRight.x < mapBounds.min.x || bottomRight.x > mapBounds.max.x ||
+         bottomRight.y < mapBounds.min.y || bottomRight.y > mapBounds.max.y ) { return false; }
+    screenToGroundPlane(topRight.x, topRight.y);
+    topRight += glm::dvec2(m_pos.x, m_pos.y);
+    if ( topRight.x < mapBounds.min.x || topRight.x > mapBounds.max.x ||
+         topRight.y < mapBounds.min.y || topRight.y > mapBounds.max.y ) { return false; }
+    screenToGroundPlane(topLeft.x, topLeft.y);
+    topLeft += glm::dvec2(m_pos.x, m_pos.y);
+    if ( topLeft.x < mapBounds.min.x || topLeft.x > mapBounds.max.x ||
+         topLeft.y < mapBounds.min.y || topLeft.y > mapBounds.max.y ) { return false; }
```

Comment:
Might be worthwhile to add a method like `bool BoundingBox::contains(glm::dvec2 point)`